test/controllers/FastlyCacheControllerTest.ts

```diff
@@ -0,0 +1,142 @@
+/// <reference path="../../typings/index.d.ts" />
+// tslint:disable-next-line:no-var-requires
+const Lab = require('lab');
+export const lab = Lab.script();
+const describe = lab.describe;
+const it = lab.it;
+const beforeEach = lab.beforeEach;
+
+import * as request from 'supertest';
+import * as nock from 'nock';
+
+import { server } from '../../src/darkstar';
+
+describe('/v1/caches/fastly', () => {
+  let fastlyAPIMock: nock.Scope;
+
+  beforeEach( (done: Function) => {
+    fastlyAPIMock = nock('https://api.fastly.com');
+    done();
+  });
+
+  describe('/zones', () => {
+    describe('DELETE', () => {
+      let flushRequest: request.Test;
+      let fastlyFlushMock: nock.Scope;
+
+      beforeEach( (done: Function) => {
+        flushRequest = request(server.listener)
+          .delete('/v1/caches/fastly/zones/1')
+          .set('accept', 'application/json');
+        fastlyFlushMock = fastlyAPIMock
+          .intercept('/*', 'PURGE');
+        done();
+      });
+
+      it('should flush a complete zone of the Fastly cache', (done: Function) => {
+        fastlyFlushMock
+          .matchHeader('accept', 'application/json')
+          .matchHeader('authorization', 'U2FsdGVkX18D8TD+GD3REqc8cdjRikR6socyNOVSrN0=')
+          .reply(200, { success: true });
```

Comment:
Sorry but your tests are not working as this is a call signature to Fasterize API, not Fastly API.